gameCode.cpp

```diff
@@ -0,0 +1,157 @@
+#include <iostream>
+#include <iosfwd>
+#include <iomanip>
+#include <cstdio>
+#include <cstring>
+#include <cstdlib>
+#include <ctime>
+#include <cmath>
+#include <cassert>
+#include <cctype>
+#include <climits>
+#include <vector>
+#include <bitset>
+#include <set>
+#include <queue>
+#include <stack>
+#include <map>
+#include <deque>
+#include <string>
+#include <list>
+#include <iterator>
+#include <sstream>
+#include <complex>
+#include <fstream>
+#include <functional>
+#include <numeric>
+#include <utility>
+#include <algorithm>
+#include <assert.h>
+#include <unordered_map>
+#include "Person.h"
+#include "Snake.h"
+#include "Ladder.h"
+using namespace std;
+
+class Game
+{
+	vector<Snake> snakesLocations;
+	vector<Ladder> laddersLocations;
+	vector<Person> players;
+	int numberOfSnakes, numberOfLadders, numberOfPlayers;
+
+public:
+
+	Game(int numSnakes, vector<Snake> snakesPos, int numLadders, vector<Ladder> laddersPos, int numPlayers, vector<Person> playersInfo)
+	{
+		snakesLocations = snakesPos;
+		laddersLocations = laddersPos;
+		players = playersInfo;
+		numberOfSnakes = numSnakes;
+		numberOfLadders = numLadders;
+		numberOfPlayers = numPlayers;
+	}
+
+	void startGame()
+	{
+		int diceValue, currentPlayerPos, earlierPlayerPos;
+
+		if (numberOfPlayers == 0)
+		{
+			cout << "The game can't be started with zero players" << endl;
+			return;
+		}
+
+		int currentPlayer = 0;  // first Player to be indexed at 0
+		currentPlayerPos = earlierPlayerPos = players[currentPlayer].getPos();
+		int snakeBite = 0, ladderFound = 0;
+		while(currentPlayerPos != 100)
+		{
+			diceValue = rand() % 6 + 1;
+			if (currentPlayerPos + diceValue > 100)
+			{
+				cout << players[currentPlayer].getName() + " cannot make a move and cross 100" << endl;
+				currentPlayer = (currentPlayer + 1) % numberOfPlayers;
+				currentPlayerPos = earlierPlayerPos = players[currentPlayer].getPos();
+				continue;
+			}
+			else
+			{
+
+				currentPlayerPos += diceValue;
+				for (int i = 0; i < numberOfSnakes; i++)
+				{
+					if (snakesLocations[i].getSecond() == currentPlayerPos)
+					{
+						currentPlayerPos = snakesLocations[i].getFirst();
+						snakeBite = 1;
+						break;
+					}
+				}
+
+				for (int i = 0; i < numberOfLadders; i++)
+				{
+					if (laddersLocations[i].getFirst() == currentPlayerPos)
+					{
+						currentPlayerPos = laddersLocations[i].getSecond();
+						ladderFound = 1;
+						break;
+					}
+				}
+			}
+			if (!snakeBite && !ladderFound)
+				cout << players[currentPlayer].getName() + " rolled a " + to_string(diceValue) + " and moved from " + to_string(earlierPlayerPos) + " to " + to_string(currentPlayerPos) << endl;
+			else if(snakeBite)
+				cout << players[currentPlayer].getName() + " rolled a " + to_string(diceValue) + " and moved from " + to_string(earlierPlayerPos) + " to " + to_string(currentPlayerPos) + " where a snake was found at " + to_string(earlierPlayerPos + diceValue) << endl;
```

Comment:
You should not deviate from the requirement. The requirement had a specified output format